extensions/amp-user-notification/0.1/amp-user-notification.js

```diff
@@ -209,20 +214,39 @@ export class AmpUserNotification extends AMP.BaseElement {
    * @return {!Promise}
    */
   postDismissEnpoint_() {
+    const enctype = this.element.getAttribute('enctype') ||
+      'application/json;charset=utf-8';
     return Services.xhrFor(this.win).fetchJson(
         dev().assertString(this.dismissHref_),
-        {
-          method: 'POST',
-          credentials: 'include',
-          requireAmpResponseSourceOrigin: false,
-          body: /** @type {!JsonObject} */({
-            'elementId': this.elementId_,
-            'ampUserId': this.ampUserId_,
-          }),
-          headers: {
-            'Content-Type': 'application/json;charset=utf-8',
-          },
-        });
+        this.buildPostDismissRequest_(enctype, this.elementId_,
+            this.ampUserId_));
+  }
+
+  /**
+   * Creates a Request to be used for postDismiss
+   * @private
+   * @param {string} enctype
+   * @param {string} elementId
+   * @param {string} ampUserId
+   * @return {!Object}
+   */
+  buildPostDismissRequest_(enctype, elementId, ampUserId) {
+    let body = dict({
+      'elementId': elementId,
+      'ampUserId': ampUserId,
+    });
+    if (enctype === FORM_DATA_ENC_TYPE) {
```

Comment:
It seems like this is [generically useful](https://github.com/ampproject/amphtml/issues/7607), so it should be extracted into `XHR#fetchJson`.

Reply:
@jridgewell Probably a good idea, but how do you see API to be here? Just a new `enctype` field in fetch's init options? And then re-encode JSON map as a query string? I'd be good with that, except we need to make it clear that not just any JSON map would work, e.g. not nested values/arrays. It's probably pretty obvious, but still...

Reply:
I was thinking just check the `Content-Type` header. We can provide extra assertions in `serializeQueryString` to ensure values are primitives.

Reply:
Good point. We actually do most of this already in `fetchJson` so that should fit nicely. Let's do!